distribution/lib/Standard/Base/0.0.0-dev/src/System/File_Format.enso

```diff
@@ -46,12 +46,9 @@ format_widget =
     all_types = [Auto_Detect] + format_types
     make_ctor type_obj =
         type_name = Meta.get_qualified_type_name type_obj
-
-        ## Temporary work around to work out if need to add the constructor name
-        is_singleton_type = type_obj==JSON_Format || (type_name.ends_with "_Format" . not)
-        if is_singleton_type then type_name else
-            simple_name = Meta.get_simple_type_name type_obj
-            "(" + type_name + "." + (simple_name.replace "_Format" "") + ")"
+        ctors = Meta.meta type_obj . constructors
+        if ctors.length == 0 then type_name else
```

Comment:
```suggestion
        ctors = Meta.meta type_obj . constructors
        is_singleton_type = ctors.length == 0
        if is_singleton_type then type_name else
```

I'd still keep the variable describing what this condition _means_. IMO it makes it easier to understand.

The point here is not how many constructors are there but what kind of type definition that is, the constructor count is just a _means_ to check that.

Reply:
ok - happy to add that.